doc/source/guidelines/optional_packages.rst

```diff
@@ -0,0 +1,248 @@
+.. _optional_packages:
+
+Handling Optional Packages
+==========================
+
+The ``setuptools``, ``flit``, and ``poetry`` build systems all allow you to
+declare dependencies that only get installed under specific circumstances. This
+makes it convenient to declare dependencies for ancillary functions such as
+"plotting", "tests", or "docs". This allows you to pragmatically integrate
+optional dependencies which can be installed as optional requirements rather
+than individual packages.
```

Comment:
```suggestion
"plotting", "tests", or "docs". You can pragmatically integrate
dependencies that can be installed as optional requirements rather
than individual packages.
```

Reply:
@akaszynski Did you mean "pragmatically integrate" or "programmatically integrate"?

Reply:
programmatically

I edited your comment to reflect that. Thanks for pointing that out.